deployment/terraform/iam.tf

```diff
@@ -192,3 +192,101 @@ resource "aws_iam_instance_profile" "batch_container_instance" {
   name = "${aws_iam_role.batch_container_instance_ec2.name}"
   role = "${aws_iam_role.batch_container_instance_ec2.name}"
 }
+
+#
+# Tilegarden executor
+#
+data "aws_iam_policy_document" "lambda_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+data "aws_iam_policy_document" "lambda_invoke" {
+  statement {
+    effect = "Allow"
+    resources = ["*"]
+    actions = [
+      "lambda:InvokeFunction",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "logs_create_and_write" {
+  statement {
+    effect = "Allow"
+    resources = ["arn:aws:logs:*:*:*"],
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "vpc_access" {
+  statement {
+    effect = "Allow"
+    resources = ["*"],
+    actions = [
+      "ec2:CreateNetworkInterface",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "s3_write_tile_cache" {
+  statement {
+    effect = "Allow"
+    resources = [
+      "arn:aws:s3:::${lower(var.environment)}-pfb-tile-cache-${var.aws_region}",
+      "arn:aws:s3:::${lower(var.environment)}-pfb-tile-cache-${var.aws_region}/*",
+    ]
+    actions = [
+      "s3:ListBucket",
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:DeleteObject",
+      "s3:GetLifecycleConfiguration",
+      "s3:PutLifecycleConfiguration",
+      "s3:GetBucketWebsite",
```

Comment:
This is nice and simple compared to the permissions we wound up setting manually! Glad you were able to pare it down to the essentials.